src/rust/inetstack/protocols/layer4/mod.rs

```diff
@@ -78,15 +98,41 @@ impl Peer {
         let udp: SharedUdpPeer = SharedUdpPeer::new(config, runtime.clone(), layer3_endpoint.clone())?;
         let tcp: SharedTcpPeer = SharedTcpPeer::new(config, runtime.clone(), layer3_endpoint.clone(), rng_seed)?;
 
-        Ok(Peer { tcp, udp })
+        Ok(Peer {
+            tcp,
+            udp,
+            layer3_endpoint,
+        })
     }
 
-    pub fn receive_tcp_packet(&mut self, src_ipv4_addr: Ipv4Addr, pkt: DemiBuffer) {
-        self.tcp.receive(src_ipv4_addr, pkt)
-    }
+    pub fn poll_once(&mut self) {
+        let batch: ArrayVec<(Ipv4Addr, IpProtocol, DemiBuffer), RECEIVE_BATCH_SIZE> = match {
+            timer!("inetstack::layer4_endpoint::poll_once");
```

Comment:
Would this essentially only measure layer3's `receive()`? Should we add the timer there then?

Reply:
This measures how long it takes the poll the underlying layer/hardware for incoming packets.

Reply:
I mean that the current timer will only measure the call to layer3.receive() and everything inside it. There's nothing else happening in the layer4 here. So it would be ok to put it in layer3. But this is ok for now.